drivers/char/rust_example.rs

```diff
@@ -49,9 +59,17 @@ impl KernelModule for RustExample {
     fn init() -> KernelResult<Self> {
         println!("Rust Example (init)");
         println!("Am I built-in? {}", !cfg!(MODULE));
-        println!("Parameters:");
-        println!("  my_bool:  {}", my_bool.read());
-        println!("  my_i32:   {}", my_i32.read());
+        {
+            let lock = THIS_MODULE.kernel_param_lock();
+            println!("Parameters:");
+            println!("  my_bool:    {}", my_bool.read());
+            println!("  my_i32:     {}", my_i32.locked_read(&lock));
+            println!(
+                "  my_str:     {:?}",
+                core::str::from_utf8(my_str.locked_read(&lock))
```

Comment:
I think we now support the `?` operator here, so the module will fail loading rather than printing the `Result`.

Reply:
done